src/test/java/edu/hm/hafner/util/SetTest.java

```diff
@@ -0,0 +1,246 @@
+        testHash.add(2401);
+        assertThat(testHash.isEmpty()).isFalse();
+
+        /* Test: Removing thing (until empty), it should getting TRUE after removing the last Object */
+        testHash.remove(511);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.remove(2106);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.remove(1310);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.remove(2401);
+        assertThat(testHash.isEmpty()).isTrue();
+
+        /* Test: Adding and removing more than one Object. Test if isEmpty() acts as expected. */
+        testHash.addAll(extendHash);
+        assertThat(testHash.isEmpty()).isFalse();
+        testHash.removeAll(extendHash);
+        assertThat(testHash.isEmpty()).isTrue();
+    }
+
+    @Test
+    void contains() {
+        testPrep();
+
+        /* Test: Positive test of contains() */
+        assertThat(extendHash.contains(511)).isTrue();
+        assertThat(extendHash.contains(2401)).isTrue();
+        for (int i = 75; i < 100; i++) {
+            assertThat(extendHash.contains(i)).isTrue();
+        }
+
+        /* Test: Negative test of contains() */
+        assertThat(extendHash.contains(3112)).isFalse();
+        assertThat(extendHash.contains(1996)).isFalse();
+        for (int i = 100; i < 150; i++) {
+            assertThat(extendHash.contains(i)).isFalse();
+        }
+    }
+
+    @Test //additional
+    void containsAll() {
+        testPrep();
+
+        /* Test: Check a filled Map against an empty one */
+        assertThat(extendHash.containsAll(testHash)).isTrue();
+        assertThat(testHash.containsAll(extendHash)).isFalse();
+
+        /* Test: Check exactly same filled Maps against each other */
+        testHash.addAll(extendHash);
+        assertThat(extendHash.containsAll(testHash)).isTrue();
+        assertThat(testHash.containsAll(extendHash)).isTrue();
+
+        /* Test: Check two different filled Maps against each other */
+        testHash.add(3112);
+        assertThat(extendHash.containsAll(testHash)).isFalse();
+        assertThat(testHash.containsAll(extendHash)).isTrue();
+        extendHash.add(2412);
+        assertThat(extendHash.containsAll(testHash)).isFalse();
+        assertThat(testHash.containsAll(extendHash)).isFalse();
+        testHash.remove(3112);
+        testHash.remove(511);
+        assertThat(extendHash.containsAll(testHash)).isTrue();
+        assertThat(testHash.containsAll(extendHash)).isFalse();
+
+        /* Test: Check the Map against itself */
+        testHash = extendHash;
+        assertThat(extendHash.containsAll(testHash)).isTrue();
+        assertThat(testHash.containsAll(extendHash)).isTrue();
+    }
+
+    @Test
+    void add() {
+        testHash = new HashSet<>();
+        extendHash = new HashSet<>();
+
+        /* Test: Adding only one Integer each time */
+        assertThat(extendHash).isEmpty();
+        assertThat(extendHash.hashCode()).isEqualTo(0);
```

Comment:
In Tests möglichst nicht mit `hashCode` arbeiten. Der Test und dessen Korrektheit ist dadurch nur schwer nachzuvollziehen. Stattdessen lieber `contains` nutzen.

Reply:
Kann ich mit contains auch alles quer durchtesten?
Ich meine der Test für die hashCode erfolgt vorher und ist nichts anderes als eine Quersumme.

Reply:
Ja. `containsAll` wäre z.B. auch eine Möglichkeit oder von assertJ gibt es auch Methoden die überprüfen ob bestimmte Elemente enthalten sind. Das ist deutlich besser nachzuvollziehen als mit dem HashCode. Ich musste zum Beispiel erst mal nachschauen, ob der HashCode eines Sets als Summe definiert ist (Das ist normalerweise unwichtig). Dazu kommt, dass die Summe nicht eindeutig auf die enthaltenen Elemente schließen lässt. Z.B.: [2, 3] => 5 und [1, 4] => 5.